examples/ex-07.c

```diff
@@ -0,0 +1,772 @@
+
+   // u1 = u0 + hf(u0)
+   VecAxpy(h, k1, u0);
+}
+
+/*--------------------------------------------------------------------------
+ * User-defined routines and structures required by Braid
+ *--------------------------------------------------------------------------*/
+
+/* can put anything in my app and name it anything as well */
+typedef struct _braid_App_struct
+{
+   MPI_Comm comm;
+   double tstart;
+   double tstop;
+   int ntime;
+   int rank;
+   double lyap_exps[VecSize]; // temporary storage for lyapunov exponents
+   FILE *file;                // saves the state vectors
+   FILE *file_lv;             // saves the Lyapunov vectors and exponents
+
+} my_App;
+
+int my_App_Init(my_App *app,
+                MPI_Comm comm_,
+                double tstart_,
+                double tstop_,
+                int ntime_,
+                int rank_,
+                FILE *file,
+                FILE *file_lv)
+{
+   (app->comm) = comm_;
+   (app->tstart) = tstart_;
+   (app->tstop) = tstop_;
+   (app->ntime) = ntime_;
+   (app->rank) = rank_;
+   (app->file) = file;
+   (app->file_lv) = file_lv;
+   for (int i = 0; i < rank_; i++)
+   {
+      (app->lyap_exps)[i] = 0.;
+   }
+
+   return 0;
+}
+
+/* can put anything in my vector and name it anything as well */
+typedef struct _braid_Vector_struct
+{
+   VEC values;
+} my_Vector;
+
+int my_Step(braid_App app,
+            braid_Vector ustop,
+            braid_Vector fstop,
+            braid_Vector u,
+            braid_StepStatus status)
+{
+   double tstart; /* current time */
+   double tstop;  /* evolve to this time */
+   braid_StepStatusGetTstartTstop(status, &tstart, &tstop);
+
+   double h; /* dt value */
+   h = tstop - tstart;
+
+   // get the number of Lyapunov vectors we need to propagate
+   int rank; /* rank of Delta correction */
+   braid_StepStatusGetDeltaRank(status, &rank);
+   MAT LinProp = {{0., 0., 0.}, {0., 0., 0.}, {0., 0., 0.}};
+
+   if (rank > 0) // we are propagating Lyapunov vectors
+   {
+      Euler((u->values), h, &LinProp);
+   }
+   else
+   {
+      Euler((u->values), h, NULL);
+   }
+
+   for (int i = 0; i < rank; i++)
+   {
+      // get a reference to the ith Lyapunov vector
+      my_Vector *psi;
+      braid_StepStatusGetBasisVec(status, &psi, i);
+
+      // propagate the vector from tstart to tstop
+      if (psi)
+      {
+         MatVec(LinProp, psi->values);
+      }
+   }
+
+   /* no refinement */
+   braid_StepStatusSetRFactor(status, 1);
+
+   return 0;
+}
+
+int my_Init(braid_App app, double t, braid_Vector *u_ptr)
+{
+   my_Vector *u;
+
+   u = (my_Vector *)malloc(sizeof(my_Vector));
+
+   /* Initial condition */
+   // u->values[0] = -1.8430428;
+   // u->values[1] = -0.07036326;
+   // u->values[2] = 23.15614636;
+   u->values[0] = 9.614521163788712;
+   u->values[1] = 17.519127090401067;
+   u->values[2] = 13.94230712158098;
+
+   *u_ptr = u;
+
+   return 0;
+}
+
+int my_InitBasis(braid_App app, double t, int index, braid_Vector *u_ptr)
```

Comment:
I would add a comment that this is a new function required for delta, and what you have to do with it